### PR TITLE
RA2 - Added Apocalypse Tank

### PR DESCRIPTION
**Pull request by MustaphaTR**
Added Apocalypse Tank. I did not try it yet and needs a squash first. I can do them when i go home. I'm at school now.

### CODE REVIEW

**Comment by Sheepsaurus** [comment]
Is it possible for outside people (Aka. Me) to try the achieved current state of the game? - I am very excited to see the end-result, and would love to help.

**Comment by MustaphaTR** [comment]
@Sheepsaurus do things below.

1. Download Both OpenRAs and RA2 mod from Download Zip button.
2. Look here to run OpenRAs lastest bleed. https://github.com/OpenRA/OpenRA/wiki/Compiling#test-your-changes
3. Unzip RA2 mod, copy it to mods folder of OpenRA.
4. Rename it as ra2.
5. Copy RA2 mix files from real RA2 folder to RA2 mods folder.
6. Run game.

File: weapons.yaml [hunk 1]
```diff
@@ -675,3 +675,116 @@ GrandCannonWeapon:
...
+		Inaccuracy: 128
+		Image: DRAGON
+		ROT: 10
+		Speed: 170
+		RangeLimit: 35
+	Warhead@1Dam: SpreadDamage
+		Spread: 171
+		Damage: 80
+		DeathType: 3
+		ProneModifier: 70
```
Comment (abcdefg30) [inline weapons.yaml]: I think this isn't needed, is it?

**Comment by abcdefg30** [comment]
Please squash the commits.

File: weapons.yaml (hunk 1 above)
Comment (MustaphaTR) [inline weapons.yaml]: You are right it is an AA weapon. I copied it from TS MammothTusk.

**Comment by MustaphaTR** [comment]
> needs a squash first. I can do them when i go home. I'm at school now.

File: rules/vehicles.yaml [hunk 2]
```diff
@@ -294,3 +294,51 @@ htnk:
...
+	Inherits: ^Vehicle
+	Valued:
+		Cost: 1750
+	Tooltip:
+		Name: Apocalypse Tank
+		Description: Soviet Advanced Double Barreled Tank.\nStrong vs. Vehicles, Aircraft\nWeak vs. Infantry
+	Buildable:
+		Queue: Vehicle
+		BuildPaletteOrder: 60
+ 		Prerequisites: ~naweap, natech
```
Comment (abcdefg30) [inline rules/vehicles.yaml]: Could you change `natech` to `~natech` while you're at it, please?

**Comment by abcdefg30** [comment]
With above done :+1: .

File: rules/vehicles.yaml (hunk 2 above)
Comment (MustaphaTR) [inline rules/vehicles.yaml]: I think it should be shown as unbuildable when we have Soviet War Factory but not Tech Center. This is better IMO.

Comment (abcdefg30) [inline rules/vehicles.yaml]: Ok, then let it as is.

File: rules/vehicles.yaml [hunk 3]
```diff
@@ -294,3 +294,51 @@ htnk:
...
 	WithVoxelTurret:
 	WithVoxelBarrel:
+
+apoc:
+	Inherits: ^Vehicle
+	Valued:
+		Cost: 1750
+	Tooltip:
+		Name: Apocalypse Tank
+		Description: Soviet Advanced Double Barreled Tank.\nStrong vs. Vehicles, Aircraft\nWeak vs. Infantry
```
Comment (Phrohdoh) [inline rules/vehicles.yaml]: I think we need something more menacing here.

File: rules/vehicles.yaml [hunk 4]
```diff
@@ -294,3 +294,51 @@ htnk:
...
+		UpgradeMinEnabledLevel: 1
+	Armament@ANTIAIR:
+		Weapon: MammothTusk
+		LocalOffset: 640,0,384
+	AttackTurreted:
+	AutoTarget:
+	RenderSprites:
+	Selectable:
+		Bounds: 64, 52, 0, -6
+		Voice: SovietVehicleVoice
```
Comment (Phrohdoh) [inline rules/vehicles.yaml]: The apoc has its own voice. We should look to add it.

**Comment by Phrohdoh** [comment]
@Sheepsaurus you only need `ra2.mix`, `language.mix`, and `theme.mix` in your Content folder now.
If you'd like to join us on IRC we can try and sort out any issues you run into.

File: rules/vehicles.yaml (hunk 4 above)
Comment (MustaphaTR) [inline rules/vehicles.yaml]: In YR. Not in RA2. It should stay as is IMO.

File: rules/vehicles.yaml (hunk 3 above)
Comment (MustaphaTR) [inline rules/vehicles.yaml]: Do you have an idea.

File: rules/vehicles.yaml (hunk 4 above)
Comment (Phrohdoh) [inline rules/vehicles.yaml]: I don't recall that, but okay.

Comment (MustaphaTR) [inline rules/vehicles.yaml]: https://github.com/MustaphaTR/oramod-ra2/blob/master/extra-files/rules.ini#L5037-L5039

File: rules/vehicles.yaml [hunk 5]
```diff
@@ -294,3 +294,51 @@ htnk:
...
 	WithVoxelTurret:
 	WithVoxelBarrel:
+
+apoc:
+	Inherits: ^Vehicle
+	Valued:
+		Cost: 1750
+	Tooltip:
+		Name: Apocalypse Tank
+		Description: Soviet Advanced Battle Tank with Double Barrel\nand Anti-Aircraft Missile Launcher.\n\nStrong vs. Vehicles, Aircraft\nWeak vs. Infantry
```
Comment (Phrohdoh) [inline rules/vehicles.yaml]: This is extremely verbose and is just filled with more adjectives.
I'll think of something later.

Comment (abcdefg30) [inline rules/vehicles.yaml]: > 21:33 (+Phrohdoh) honestly I’m all for dropping Description as I think it’s dumb
21:33 (+Phrohdoh) but I can’t come up with one so go ahead and merge if it works

**Comment by abcdefg30** [comment]
Looks still good to me. :+1:

**Comment by Phrohdoh** [comment]
I guess this wasn't tested well enough. #57

**Pull request merged**